### ISSUE TITLE
Pylint false report

### ISSUE THREAD

**Issue by matejsp** [pylint-dev/pylint#9134]
### Bug description

```python
import typing

from marshmallow import fields
from marshmallow import validate


class OrderAction:

    def validate(self):
        pass


class AdminCancelOrders(OrderAction):
    order_ids: typing.List[int] = fields.List(fields.Int(), required=True, validate=validate.Length(min=1))
    remote_ip: str = fields.String(required=True)

    def validate(self):
        super(AdminCancelOrders, self).validate()
```
```


### Configuration

_No response_

### Command used

```shell
pylint pylint_bug_test.py
```

### Pylint output

```python
pylint pylint_bug_test.py
************* Module pylint_bug_test
pylint_bug_test.py:14:84: E0601: Using variable 'validate' before assignment (used-before-assignment)
```

### Expected behavior

It should report that validate is used before assignement as it did not in 2.7.x

### Pylint version

```shell
pylint 3.0.1
astroid 3.0.0
Python 3.11.2 (main, Mar 10 2023, 07:05:07) [Clang 14.0.0 (clang-1400.0.29.202
```

### OS / Environment

MacOSX Ventura

### Additional dependencies

```python
marshmallow==2.21.0.2
```

**Comment by nickdrozd** [comment]
Similar to https://github.com/pylint-dev/pylint/issues/9115

**Comment by matejsp** [comment]
I saw that one false positive was solved with 3.0.2.
```
Fix used-before-assignment false positive for generic type syntax (PEP 695, Python 3.12).
Closes #9110 ([#9110](https://github.com/pylint-dev/pylint/issues/9110))
```
But unfourtenately does not work for this case. Still the same reported issue.

pylint = "==3.0.2"
astroid = "==3.0.1"
